reframe/core/logging.py

```diff
@@ -284,13 +298,22 @@ def __init__(self, name, level=logging.NOTSET):
         # class' check
         super().__init__(name, logging.NOTSET)
         self.level = _check_level(level)
+        self._stream_handlers = []
 
     def __repr__(self):
         return debug.repr(self)
 
     def setLevel(self, level):
         self.level = _check_level(level)
 
+    def addHandler(self, hdlr):
+        if isinstance(hdlr, StreamHandler):
+            self._stream_handlers.append(hdlr)
```

Comment:
Even if the `Logger.handlers` attribute is not documented, I think we can make use of it, since it's public. We do this already. This would make the whole implementation much more straightforward.